### PR TITLE
[internal-enrichment] Add Virustotal Downloader Connector 

### PR DESCRIPTION
**Pull request by YungBinary**
### Proposed changes

* [internal-enrichment] Add Virustotal Downloader Connector 

### Checklist

- [x] I consider the submitted work as finished
- [x] I tested the code for its functionality using different use cases
- [x] I added/update the relevant documentation (either on github or on notion)
- [x] Where necessary I refactored code to improve the overall quality

### Further comments

This Connector introduces the ability for automated and manual hash checks in Virustotal to download associated Artifact contents and upload them to OpenCTI. A relationship will be made between a StixFile and the uploaded Artifact. If the file does not exist in Virustotal, an error will show letting you know.

### CODE REVIEW

File: internal-enrichment/virustotal-downloader/src/virustotal-downloader.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+        except Exception as e:
+            raise Exception(
+                f"Failed to download/upload Artifact with hash {hash_value}, exception: {e}"
+            )
+
+        # Create a relationship between the StixFile and the new Artifact
+        relationship = Relationship(
+            id=OpenCTIStix2Utils.generate_random_stix_id("relationship"),
+            relationship_type="related-to",
```
Comment (nor3th) [inline internal-enrichment/virustotal-downloader/src/virustotal-downloader.py:75]: Should we in the future (once this PR is merged in the platform https://github.com/OpenCTI-Platform/opencti/pull/1803/files#diff-8f804f0d600cb54e6d42acc99961742916a240a86e8281cac53e41ccb38e16c5R893 ) to change the relationship from `related-to` to Artifact - `content` -> StixFile. As specified here:

content_ref (optional) | identifier | Specifies the content of the file, represented as an Artifact object. The object referenced in this property MUST be of type artifact.
-- | -- | --

https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_vq03pryd7u32

Or should we leave it as it is as a simple `related-to` relationship?

Comment (nor3th) [inline internal-enrichment/virustotal-downloader/src/virustotal-downloader.py:75]: Should we in the future (once this PR is merged in the platform https://github.com/OpenCTI-Platform/opencti/pull/1803/files#diff-8f804f0d600cb54e6d42acc99961742916a240a86e8281cac53e41ccb38e16c5R893 ) to change the relationship from `related-to` to Artifact - `content` -> StixFile. As specified here:

content_ref (optional) | identifier | Specifies the content of the file, represented as an Artifact object. The object referenced in this property MUST be of type artifact.
-- | -- | --

https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_vq03pryd7u32

Or should we leave it as it is as a simple `related-to` relationship?

**Review by nor3th — CHANGES_REQUESTED** [review]
Could you also please add your connector to the CircleCI config? See as example: https://github.com/OpenCTI-Platform/connectors/commit/d317213f6e54e473cceb8fe191a297f373da7edc#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47

**Comment by YungBinary** [comment]
> Could you also please add your connector to the CircleCI config? See as example: [d317213#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47](https://github.com/OpenCTI-Platform/connectors/commit/d317213f6e54e473cceb8fe191a297f373da7edc#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47)

Thank you! Those changes are done.

I also went and made an issue for updating the connector dev documentation here: https://github.com/OpenCTI-Platform/connectors/issues/609

File: internal-enrichment/virustotal-downloader/src/virustotal-downloader.py (hunk 1 above)
Comment (YungBinary) [reply internal-enrichment/virustotal-downloader/src/virustotal-downloader.py:75]: I'm going to work on this right now to see what the behavior is like to see if it's wanted or not. Thanks for the tip!

**Comment by nor3th** [comment]
> > Could you also please add your connector to the CircleCI config? See as example: [d317213#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47](https://github.com/OpenCTI-Platform/connectors/commit/d317213f6e54e473cceb8fe191a297f373da7edc#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47)
> 
> Thank you! Those changes are done.
> 
> I also went and made an issue for updating the connector dev documentation here: #609

Hahah great, I hoped to avoid having to update the documentation and simply expect it to somehow become common knowledge :sweat_smile:

**Review by nor3th — APPROVED** [review]

File: internal-enrichment/virustotal-downloader/src/virustotal-downloader.py (hunk 1 above)
Comment (YungBinary) [reply internal-enrichment/virustotal-downloader/src/virustotal-downloader.py:75]: Will keep this in mind and make this change after the PR you sent me is merged.

Comment (nor3th) [reply internal-enrichment/virustotal-downloader/src/virustotal-downloader.py:75]: Sounds good ;)

**Pull request merged**